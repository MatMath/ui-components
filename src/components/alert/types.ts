export enum AlertSeverity {
  SUCCESS = 'success',
  ERROR = 'error'
}

export enum AlertSize {
  SMALL = 'small',
  MEDIUM = 'medium'
}
