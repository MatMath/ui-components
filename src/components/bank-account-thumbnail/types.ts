export type BankAccountThumbnailSize = 'small' | 'medium';
