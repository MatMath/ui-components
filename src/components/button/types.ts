export type ButtonNature = 'primary' | 'secondary' | 'danger' | 'ghost';
export type ButtonSize = 'small' | 'medium' | 'large';
export type ButtonTheme = 'light' | 'dark' | 'warning' | 'alert';
