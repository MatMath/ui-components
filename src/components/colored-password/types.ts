export type StringCategory = 'letter' | 'number' | 'special';

export interface CategorizedString {
  text: string;
  category: StringCategory;
}
