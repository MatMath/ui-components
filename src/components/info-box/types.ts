export const InfoBoxSeverity = {
  ALERT: 'alert',
  STRONG: 'strong',
  SUBTLE: 'subtle',
  WARNING: 'warning'
} as const;

export const InfoBoxSize = {
  SMALL: 'small',
  SIMPLE: 'simple',
  DESCRIPTIVE: 'descriptive'
} as const;
