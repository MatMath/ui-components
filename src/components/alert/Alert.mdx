---
name: Alert
menu: Components
---

import { Playground, Props } from 'docz';
import { Alert } from './Alert';

# Alert

An alert component

## Properties

<Props of={Alert} />

## Usage

### Basic usage

Examples below show usage of the Alert

<Playground
  style={{
    preview: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'flex-start'
    }
  }}
>
  <div style={{ marginBottom: '10px'}}>
    <Alert severity="success">This is a success message!</Alert>
  </div>

<div style={{ marginBottom: '10px' }}>
  <Alert severity='success' showCloseIcon={true}>
    This is a success message!
  </Alert>
</div>

<div style={{ marginBottom: '10px' }}>
  <Alert severity='success' showIcon={false}>
    This is a success message!
  </Alert>
</div>

<div style={{ marginBottom: '10px' }}>
  <Alert severity='error' showCloseIcon={true}>
    Something went wrong.
  </Alert>
</div>

</Playground>
