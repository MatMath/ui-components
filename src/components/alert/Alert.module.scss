@use '../../colors/colors';
@use '../../typography/typography';

.root {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  height: 60px;
  align-items: center;
  border-radius: 4px;
}

.root.success {
  background-color: colors.$accessible-validator-green;
}

.root.error {
  background-color: colors.$accessible-validator-red;
}

.content {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
}

.description {
  @include typography.body-small;
  color: colors.$white;
  font-size: 18px;
  line-height: 20px;
  padding-left: 20px;
}

.icon {
  padding-left: 20px;
}

.closeIcon {
  padding-left: 20px;
  padding-right: 20px;

  &:hover:not(:disabled), &:focus:not(:disabled) {
    border: 1px solid transparent;
    background-color: transparent;
  }
}
