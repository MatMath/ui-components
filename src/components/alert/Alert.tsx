import * as React from 'react';

import {
  CheckCircleIcon,
  CloseIcon,
  CrossCircleIcon
} from '@components/icons/Icons';

import { AlertSeverity } from './types';
import { Button } from '@components/button/Button';
import { getClassNames } from '@utility/cssUtils';
import styles from './Alert.module.scss';

export interface AlertProps {
  severity?: AlertSeverity;
  showIcon?: boolean;
  showCloseIcon?: boolean;
  children: string;
}

export const Alert = ({
  severity = AlertSeverity.SUCCESS,
  showIcon = true,
  showCloseIcon = false,
  children
}: AlertProps) => {
  const [close, setClose] = React.useState(false);

  if (!children || close) {
    return null;
  }

  const getIconBySeverity = (alertSeverity: AlertSeverity) => {
    switch (alertSeverity) {
      case AlertSeverity.ERROR:
        return <CrossCircleIcon size={30} color='white' />;
      case AlertSeverity.SUCCESS:
      default:
        return <CheckCircleIcon size={30} color='white' />;
    }
  };

  const renderIcon = (alertSeverity: AlertSeverity) => {
    const icon = getIconBySeverity(alertSeverity);
    return <div className={styles.icon}>{icon}</div>;
  };

  return (
    <div className={getClassNames(styles.root, styles[severity])}>
      <div className={styles.content}>
        {showIcon && renderIcon(severity)}
        <p className={styles.description}>{children}</p>
      </div>
      {showCloseIcon && (
        <Button
          nature='ghost'
          size='small'
          className={styles.closeIcon}
          onClick={() => setClose(true)}
        >
          <CloseIcon size={20} color='white' />
        </Button>
      )}
    </div>
  );
};
