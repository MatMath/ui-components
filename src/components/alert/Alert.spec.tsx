import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import { Alert, AlertProps } from '@components/alert/Alert';
import { AlertSeverity } from '@components/alert/types';
import { CheckCircleIcon, CloseIcon } from '@components';

describe('<Alert />', () => {
  let wrapper: ReactWrapper;

  const createWrapper = ({
    severity,
    showIcon,
    showCloseIcon,
    children
  }: AlertProps) =>
    mount(
      <Alert
        severity={severity}
        showIcon={showIcon}
        showCloseIcon={showCloseIcon}
      >
        {children}
      </Alert>
    );

  it('should create an alert with success and with icon closed if children is defined', () => {
    const children = 'Success test';
    wrapper = createWrapper({
      children
    });
    expect(wrapper.exists('.root.success')).toEqual(true);
  });

  it('should not show alert if children is empty', () => {
    wrapper = createWrapper({ children: '' });
    expect(wrapper.exists('.root.success')).toEqual(false);
  });

  it('should be able to create a success alert', () => {
    const children = 'Success test';
    wrapper = createWrapper({
      severity: AlertSeverity.SUCCESS,
      children
    });
    expect(wrapper.exists('.root.success')).toEqual(true);
  });

  it('should be able to create an error alert', () => {
    const children = 'Error test';
    wrapper = createWrapper({
      severity: AlertSeverity.ERROR,
      children
    });
    expect(wrapper.exists('.root.error')).toEqual(true);
  });

  it('should close alert when close button is pressed', () => {
    const children = 'Success test';
    wrapper = createWrapper({
      severity: AlertSeverity.SUCCESS,
      showCloseIcon: true,
      children
    });
    expect(wrapper.exists('.root.success')).toEqual(true);
    const closeBtn = wrapper.find(CloseIcon);
    closeBtn.simulate('click');
    expect(wrapper.exists('.root.success')).toEqual(false);
  });

  it('should not show icon if showIcon is false', () => {
    const children = 'Success test';
    wrapper = createWrapper({
      severity: AlertSeverity.SUCCESS,
      showIcon: false,
      showCloseIcon: true,
      children
    });
    expect(wrapper.exists('.root.success')).toEqual(true);
    expect(wrapper.exists(CheckCircleIcon)).toEqual(false);
  });

  it('should not show close button if showCloseIcon is false', () => {
    const children = 'Success test';
    wrapper = createWrapper({
      severity: AlertSeverity.SUCCESS,
      showIcon: true,
      showCloseIcon: false,
      children
    });
    expect(wrapper.exists('.root.success')).toEqual(true);
    expect(wrapper.exists(CloseIcon)).toEqual(false);
  });
});
