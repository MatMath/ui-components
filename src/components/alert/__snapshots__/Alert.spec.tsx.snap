// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Should create a small alert 1`] = `
<div>
  <div
    class="root success small"
  >
    <div
      class="content"
    >
      <div
        class="icon small"
      >
        <svg
          class="svgWrapper animated"
          version="1.1"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9.23989 10.7168L12.1705 7.68797L13.2485 8.73101L9.22458 12.8898L6.97226 10.4954L8.06485 9.46763L9.23989 10.7168Z"
            fill-rule="evenodd"
          />
          <path
            d="M13.8347 15.2489C12.7597 16.0356 11.4341 16.5 10 16.5C6.41015 16.5 3.5 13.5899 3.5 10C3.5 6.41015 6.41015 3.5 10 3.5C13.5899 3.5 16.5 6.41015 16.5 10C16.5 11.4341 16.0356 12.7597 15.2489 13.8347L16.3199 14.9056C17.3729 13.551 18 11.8487 18 10C18 5.58172 14.4183 2 10 2C5.58172 2 2 5.58172 2 10C2 14.4183 5.58172 18 10 18C11.8487 18 13.551 17.3729 14.9056 16.3199L13.8347 15.2489Z"
          />
        </svg>
      </div>
      <p
        class="description small"
      >
        Success test
      </p>
    </div>
  </div>
</div>
`;

exports[`<Alert /> should be able to create an error alert 1`] = `
<div>
  <div
    class="root error medium"
  >
    <div
      class="content"
    >
      <div
        class="icon medium"
      >
        <svg
          class="svgWrapper animated"
          version="1.1"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.6637 8.39704L8.39703 12.6637L7.33637 11.603L11.603 7.33638L12.6637 8.39704Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M11.603 12.6637L7.3363 8.39704L8.39696 7.33638L12.6636 11.603L11.603 12.6637Z"
            fill-rule="evenodd"
          />
          <path
            d="M15.262 13.8167C16.0407 12.745 16.5 11.4261 16.5 10C16.5 6.41015 13.5899 3.5 10 3.5C6.41015 3.5 3.5 6.41015 3.5 10C3.5 13.5899 6.41015 16.5 10 16.5C11.4421 16.5 12.7745 16.0304 13.8526 15.2357L14.9231 16.3062C13.5656 17.3675 11.8567 18 10 18C5.58172 18 2 14.4183 2 10C2 5.58172 5.58172 2 10 2C14.4183 2 18 5.58172 18 10C18 11.8407 17.3783 13.5363 16.3334 14.8881L15.262 13.8167Z"
          />
        </svg>
      </div>
      <p
        class="description medium"
      >
        Error test
      </p>
    </div>
  </div>
</div>
`;

exports[`<Alert /> should create an alert with success, medium, and with icon closed if children is defined 1`] = `
<div>
  <div
    class="root success medium"
  >
    <div
      class="content"
    >
      <div
        class="icon medium"
      >
        <svg
          class="svgWrapper animated"
          version="1.1"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9.23989 10.7168L12.1705 7.68797L13.2485 8.73101L9.22458 12.8898L6.97226 10.4954L8.06485 9.46763L9.23989 10.7168Z"
            fill-rule="evenodd"
          />
          <path
            d="M13.8347 15.2489C12.7597 16.0356 11.4341 16.5 10 16.5C6.41015 16.5 3.5 13.5899 3.5 10C3.5 6.41015 6.41015 3.5 10 3.5C13.5899 3.5 16.5 6.41015 16.5 10C16.5 11.4341 16.0356 12.7597 15.2489 13.8347L16.3199 14.9056C17.3729 13.551 18 11.8487 18 10C18 5.58172 14.4183 2 10 2C5.58172 2 2 5.58172 2 10C2 14.4183 5.58172 18 10 18C11.8487 18 13.551 17.3729 14.9056 16.3199L13.8347 15.2489Z"
          />
        </svg>
      </div>
      <p
        class="description medium"
      >
        Success test
      </p>
    </div>
  </div>
</div>
`;

exports[`<Alert /> should not show alert if children is empty 1`] = `<div />`;

exports[`<Alert /> should not show close button if showCloseIcon is false 1`] = `
<div>
  <div
    class="root success medium"
  >
    <div
      class="content"
    >
      <div
        class="icon medium"
      >
        <svg
          class="svgWrapper animated"
          version="1.1"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9.23989 10.7168L12.1705 7.68797L13.2485 8.73101L9.22458 12.8898L6.97226 10.4954L8.06485 9.46763L9.23989 10.7168Z"
            fill-rule="evenodd"
          />
          <path
            d="M13.8347 15.2489C12.7597 16.0356 11.4341 16.5 10 16.5C6.41015 16.5 3.5 13.5899 3.5 10C3.5 6.41015 6.41015 3.5 10 3.5C13.5899 3.5 16.5 6.41015 16.5 10C16.5 11.4341 16.0356 12.7597 15.2489 13.8347L16.3199 14.9056C17.3729 13.551 18 11.8487 18 10C18 5.58172 14.4183 2 10 2C5.58172 2 2 5.58172 2 10C2 14.4183 5.58172 18 10 18C11.8487 18 13.551 17.3729 14.9056 16.3199L13.8347 15.2489Z"
          />
        </svg>
      </div>
      <p
        class="description medium"
      >
        Success test
      </p>
    </div>
  </div>
</div>
`;

exports[`<Alert /> should not show icon if showIcon is false 1`] = `
<div>
  <div
    class="root success medium"
  >
    <div
      class="content"
    >
      <p
        class="description medium"
      >
        Success test
      </p>
    </div>
  </div>
</div>
`;
